### PR TITLE
Raise ValueError for Profiler + Auto Grad Accum

### PR DESCRIPTION
**Pull request by mvpatel2000**
As title. It's very complicated to support profiler and auto grad accum since profiler requires all "close" events, eg `AFTER_X`, run. This is hard to do and not worth it right now.

### CODE REVIEW

File: composer/trainer/trainer.py [hunk 1]
```diff
@@ -141,8 +141,12 @@ def _set_evaluator_interval_and_subset_num_batches(
             evaluator.eval_interval = eval_interval
 
 
-def _is_adaptive_grad_accum(grad_accum: Union[int, str], device: Device):
+def _is_adaptive_grad_accum(grad_accum: Union[int, str], device: Device, profiler: Optional[Profiler]):
     if grad_accum == 'auto':
```
Comment (karan6181) [inline composer/trainer/trainer.py:145]: [Optional]: Though it is not necessary as part of this PR, but just wondering to add a method comments to understand the code better. Thanks!

**Review by karan6181 — APPROVED** [review]
LGTM. Thank you for your contribution!

File: composer/trainer/trainer.py [hunk 2]
```diff
@@ -141,8 +141,13 @@ def _set_evaluator_interval_and_subset_num_batches(
             evaluator.eval_interval = eval_interval
 
 
-def _is_adaptive_grad_accum(grad_accum: Union[int, str], device: Device):
+def _is_adaptive_grad_accum(grad_accum: Union[int, str], device: Device, profiler: Optional[Profiler]):
```
Comment (hanlint) [inline composer/trainer/trainer.py:144]: seems a bit odd that this method returns whether something is grad accum, but then has this weird side effect of checking the profiler. Can we just add this into the main trainer init, and accept the duplication?

File: composer/trainer/trainer.py [hunk 3]
```diff
@@ -142,6 +142,7 @@ def _set_evaluator_interval_and_subset_num_batches(
 
 
 def _is_adaptive_grad_accum(grad_accum: Union[int, str], device: Device):
+    """Checks if `grad_accum='auto'` and if its compatible with current run configurations."""
```
Comment (hanlint) [inline composer/trainer/trainer.py:145]:
```suggestion
```

**Review by hanlint — APPROVED** [review]

**Pull request merged**